packages/toolpad-core/src/runtime.tsx

```diff
@@ -29,11 +29,8 @@ export const NodeRuntimeContext = React.createContext<string | null>(null);
 // React elements to the fibers.
 interface SlotsWrapperProps {
   children?: React.ReactNode;
-  // eslint-disable-next-line react/no-unused-prop-types
```

Comment:
This is related to the update of `@mui/monorepo`. I don't know why, but it seems to be a good thing. Maybe they changed the default eslint rules.

Reply:
🤔 I'm no sure, these properties are deliberately not used in a way that is not statically analyzable. This looks more like the rule got inadvertently disabled.

Trying to add the following in the file as a test, and it doesn't seem to catch the unused prop, it does on master.
```tsx

interface Props {
  firstname: string;
  middlename: string; // middlename is never used by the Hello component
  lastname: string;
};

export function Hello({ firstname, lastname }: Props) {
  return (
    <div>
      Hello {firstname} {lastname}
    </div>
  );
}
```

Perhaps something with us importing the overrides in eslintrc? I removed it in https://github.com/mui/mui-toolpad/pull/835

Reply:
Updating the branch to verify this

Reply:
I have seen the same in https://github.com/mui/mui-toolpad/pull/809

Reply:
Just learnt about `eslint --print-config` option. The definitely rule is configured differently for master:

```json
    "react/no-unused-prop-types": [
      "error",
      {
        "customValidators": [],
        "skipShapeProps": true
      }
    ],
```

vs this PR

```json

    "react/no-unused-prop-types": [
      "off",
      {
        "customValidators": [],
        "skipShapeProps": true
      }
    ],
```

Looks like it got turned off in monorepo eslintrc: https://github.com/mui/material-ui/pull/33027/files#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5R140
If it was disabled on purpose then we can probably re-enable it for Toolpad for now.

Reply:
Ok great, so I think that we should revert this change in the main repo.

Reply:
Perhaps we can turn the rule off in javascript files only? ideally for toolpad I'd love to disallow `.propTypes` and rely on typescipt only, but that [doesn't seem possible right now](https://github.com/jsx-eslint/eslint-plugin-react/issues/3107).